### PR TITLE
Wait for CRD discovery to be successful once before reporting success

### PR DESCRIPTION
**Pull request by soltysh**
**What type of PR is this?**
/kind bug

**Special notes for your reviewer**:
/assign @sttts

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -223,6 +227,55 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
...
+			_, serverGroupsAndResources, discoveryErr := crdClient.Discovery().ServerGroupsAndResources()
+			if discoveryErr != nil {
+				klog.V(2).Info(discoveryErr)
+			}
+
+			serverCRDs, err := s.Informers.Apiextensions().InternalVersion().CustomResourceDefinitions().Lister().List(labels.Everything())
+			if err != nil {
+				return false, err
+			}
+			crdGroupsAndResources := sets.NewString()
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go]: nit: activeCRDNameVersionGroups

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go [hunk 2]
```diff
@@ -223,6 +227,55 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
...
+						continue
+					}
+					crdGroupsAndResources.Insert(fmt.Sprintf("%s.%s.%s", crd.Spec.Names.Plural, version.Name, crd.Spec.Group))
+				}
+			}
+
+			discoveryGroupsAndResources := sets.NewString()
+			for _, resourceList := range serverGroupsAndResources {
+				for _, apiResource := range resourceList.APIResources {
+					group, version := splitGroupVersion(resourceList.GroupVersion)
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go]: can't we use a syntax above which avoid this split?

**Comment by soltysh** [comment]
Reopened in https://github.com/kubernetes/kubernetes/pull/85851

**Pull request closed without merging**